File: Kurs C++/Zad6/cosinus.cpp
//
// Created by Mateusz Pater on 08.04.2016.
//

#include "cosinus.h"

double cosinus::oblicz() {
    return cos(a->oblicz());
}

cosinus::cosinus(wyrazenie *a){
    this->a = a;
}

std::string cosinus::formatuj_wyrazenie(){
    return "cos(" + a->formatuj_wyrazenie() + ")";
}

File: Kurs Java/Zadanie 3/src/com/company/Cosinus.java
/*
 * Copyright (c) 2016. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
 * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
 * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
 * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
 * Vestibulum commodo. Ut rhoncus gravida arcu.
 */

package com.company;

/**
 * author @pater
 */
public class Cosinus extends Operator1Arg {

    public Cosinus(Wyrazenie a){
        super(a);
    }

    @Override
    public double oblicz() {
        return Math.cos(a.oblicz());
    }
    @Override
    public String toString() {
        return "cos("+a+")";
    }

}
